convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,113 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+import os
+import re
+
+from convert2rhel import actions, systeminfo
+from convert2rhel.systeminfo import system_info
+
+
+logger = logging.getLogger(__name__)
+
+#:
+FIREWALLD_CONFIG_FILE = "/etc/firewalld/firewalld.conf"
+
+#:
+CLEANUP_MODULES_ON_EXIT_REGEX = re.compile(r"CleanupModulesOnExit\s*=\s*(yes|true)")
+
+
+def _is_modules_cleanup_enabled():
+    """Verify firewalld modules cleanup config is enabled.
+
+    :returns: Whether or not the CleanupModulesOnExit is set to true in
+        firewalld config.
+    :rtype: bool
+    """
+    # Return false if the config file does not exist. Either it means that
+    # firewalld is not installed or the config file was removed.
+    if not os.path.exists(FIREWALLD_CONFIG_FILE):
+        return False
+
+    contents = ""
+    with open(FIREWALLD_CONFIG_FILE, mode="r") as handler:
+        contents = [line.strip() for line in handler.readlines() if line.strip()]
+
+    # Contents list is empty for some reason, better to assume that there
+    # is no content in the file that was read.
+    if not contents:
+        return False
+
+    # If the config file has this option set to true/yes, then we need to
+    # return True to ask the user to change it to self.
+    if list(filter(CLEANUP_MODULES_ON_EXIT_REGEX.match, contents)):
+        logger.debug("Found CleanupModulesOnExit option in firewalld.conf")
+        return True
+
+    logger.debug(
+        "Couldn't find CleanupModulesOnExit in firewalld.conf. Treating it as enabled because of default behavior."
+    )
+    # Return true anyway at the end even if the option was not found as the
+    # default value for CleanupModulesOnExit in the firewalld.conf is to be
+    # enabled always.
+    return True
+
+
+class CheckFirewalldAvailability(actions.Action):
+    id = "CHECK_FIREWALLD_AVAILABILITY"
+
+    def run(self):
+        """Error out if the firewalld service is running on the system."""
+        super(CheckFirewalldAvailability, self).run()
+        logger.task("Prepare: Check that firewalld is running")
+
+        if system_info.id == "oracle" and (system_info.version.major == 8 and system_info.version.minor >= 8):
+            if systeminfo.is_systemd_managed_service_running("firewalld"):
+                if _is_modules_cleanup_enabled():
```

Comment:
```suggestion
        if system_info.id != "oracle" and system_info.version.major != 8 and system_info.version.minor < 8:
            logger.info("Skipping the check as it is relevant only for Oracle Linux 8.8 and above.")
            return
        if not systeminfo.is_systemd_managed_service_running("firewalld"):
            logger.info("Firewalld service reported that it is not running.")
            return
        if _is_modules_cleanup_enabled():
```
Just a style nitpick - not going neck deep into nested conditions.